leader-election-impl/src/main/java/com/palantir/paxos/VerifyingPaxosStateLog.java

```diff
@@ -117,13 +141,15 @@ public void writeRound(long seq, V round) {
         lock.readLock().lock();
         try {
             byte[] result = currentLog.readRound(seq);
-            byte[] experimentalResult = safeReadFromExperimental(psl -> psl.readRound(seq));
-            if (!Arrays.equals(result, experimentalResult)) {
-                log.error("Mismatch in reading round for sequence number {} between legacy and current "
-                        + "implementations. Legacy result {}, current result {}.",
-                        SafeArg.of("sequence", seq),
-                        UnsafeArg.of("legacy", hydrateIfNotNull(result)),
-                        UnsafeArg.of("current", hydrateIfNotNull(experimentalResult)));
+            if (firstEntry.isDone() && Futures.getUnchecked(firstEntry) <= seq) {
```

Comment:
In hindsight, following a suggestion from @j-baker, it might be worth not doing this check actually, since this would be a good way to test a hypothesis that the old entries aren't actually particularly important. It might still be worth keeping track of the first entry and dumping that as an arg in the result.